src/UserDataCollection.cc

```diff
@@ -0,0 +1,54 @@
+#include "podio/UserDataCollection.h"
+#include "podio/CollectionBufferFactory.h"
+#include "podio/CollectionBuffers.h"
+
+#include <tuple>
+#include <vector>
+
+namespace podio {
+
+namespace {
+  /**
+   * Helper function to register a UserDataCollection to the
+   * CollectionBufferFactory. Takes the BasicType as template argument.
+   *
+   * Returns an integer so that it can be used with std::apply
+   */
+  template <typename T>
+  int registerUserDataCollection(T) {
+    // Register with schema version 1 to allow for potential changes
+    CollectionBufferFactory::mutInstance().registerCreationFunc(
+        userDataCollTypeName<T>(), UserDataCollection<T>::schemaVersion, [](bool) {
+          return podio::CollectionReadBuffers{new std::vector<T>(), nullptr, nullptr,
+                                              [](podio::CollectionReadBuffers buffers, bool) {
+                                                return std::make_unique<UserDataCollection<T>>(
+                                                    std::move(*buffers.dataAsVector<T>()));
+                                              },
+                                              [](podio::CollectionReadBuffers& buffers) {
+                                                buffers.data = podio::CollectionWriteBuffers::asVector<T>(buffers.data);
+                                              }};
+        });
```

Comment:
This does not put the correct `schemaVersion` into the created buffers.

Reply:
This doesn't apply yet and needs to be part of #413

Reply:
sorry, your comments confuse me a bit

Reply:
Sorry. I thought I forgot to do this here while working on the other end of schema evolution in #413. But I then realized the necessary `schemaVersion` field is only introduced there, so I can do everything I need in that. Is there anything missing for this or should we merge?

Reply:
Sounds fine to me.